extensions/message-edit.md

```diff
@@ -0,0 +1,125 @@
+
+## Server implementation considerations
+
+This section is non-normative
+
+A key motivation for specifying this capability as a server tag, rather than a client-only message tag, is to enable more granular editing and deletion permissions. Clients might be able to determine which messages are their own, but other use cases would not be feasible without server validation.
+
+Such use cases might include:
+
+* Allowing channel moderators or server admins to delete unwelcome messages from others
+* Specifying a cut-off time after which message edits are no longer allowed
+
+### Message validation
+
+To implement validation, servers require a mechanism for determining the permissions of a particular edit or delete action. The user requesting the action would need to be compared against properties of the message, given only the message ID and target.
+
+Servers with message history storage could look up the message properties from the ID, but this might not be possible or desirable in all cases. Another mechanism could involve encoding any required properties within the message ID itself, e.g. the account ID, timestamp, etc. Servers might choose to encrypt this information if it isn't usually public facing. Any information encoded in a message ID is still opaque and not intended to be parsed by clients.
+
+### Fallback
+
+Server implementations might choose to inform clients that haven't negotiated the capability that an edit or deletion has taken place. The fallback method used (if any) is left up to server implementations, but could take the form of a standard NOTICE or PRIVMSG with information about the action. It might be preferable to use relative time descriptions if referring to messages in the past, for example:
+
+    :irc.example.com NOTICE #channel :nickname edited a message from 5 seconds ago: an example
+
+Implementations might also choose not to send a fallback, if this behaviour is considered too noisy for users.
+
+## Security considerations
+
+The ability to edit and delete messages does not offer any information or operational security guarantees. Once a message has been sent, assume that it will remain visible to any recipients or servers, whether or not it is subsequently edited or deleted. Above all else, clients that do not support this specification will not see any changes to the original message.
+
+## Examples
+
+Editing a message:
+
+    C: PRIVMSG #channel :anex ample
+    S: @msgid=123 :nick!u@h PRIVMSG #channel :anex ample
+    C: @draft/target-msgid=123;draft/edit-text=an\sexample EDIT #channel
+    S: @msgid=124;draft/target-msgid=123;draft/edit-text=an\sexample :nick!u@h EDIT #channel
+
+Deleting a message:
+
+    C: PRIVMSG #channel :an example
+    S: @msgid=123 :nick!u@h PRIVMSG #channel :an example
+    C: @draft/target-msgid=123 DELETE #channel
+    S: @msgid=124;draft/target-msgid=123 :nick!u@h DELETE #channel
```

Comment:
Would this be possible?


```suggestion
    S: @msgid=124;draft/target-msgid=123 :nick!u@h DELETE #channel

Deleting a TAGMSG:

    C: @draft/react=🤞TAGMSG #channel
    S: @msgid=123;draft/react=🤞TAGMSG #channel
    C: @draft/target-msgid=123 DELETE #channel
    S: @msgid=124;draft/target-msgid=123 :nick@u@h DELETE #channel
```

Reply:
Possibly. I was also considering just proposing an unreact tag a while ago. Any other TAGMSGs you'd want to delete?

Reply:
At this point no, there isn't really many other designed tags to consider. At this point I think only `typing` from https://ircv3.net/irc/ applys to `TAGMSG`. `typing` already designed its own mechanism for this (with additional semantics via `paused` and `done`), although perhaps it would have fit nicely into a single DELETE. Are there any other tag ideas in-flight that we might want to consider here?

I've definately see how useful unreact can be in other protocols, temporary reacting when something is in progress, in review and wanting to remove afterwards. There's also the element of removing an accidental tap or click such as if a client UI offers a single tap/click on top of an existing reaction. I agree it could be solved another way too.

You may also encode a reaction in a PRIVMSG, and then you'd be able to use `DELETE` on it.